cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs

```diff
@@ -72,6 +73,22 @@ prefixFlag prefix longFlag =
     Nothing -> longFlag
     Just prefix' -> prefix' <> "-" <> longFlag
 
+bounded :: forall a. (Bounded a, Integral a, Show a) => String -> Opt.ReadM a
+bounded t = Opt.eitherReader $ \s -> do
+  i <- Read.readEither @Integer s
+  when (i < fromIntegral (minBound @a)) $ Left $ t <> " must not be less than " <> show (minBound @a)
+  when (i > fromIntegral (maxBound @a)) $ Left $ t <> " must not greater than " <> show (maxBound @a)
```

Comment:
Wait, those conditions are not correct. `minBound` and `maxBound` are valid values of `a`, for example: `minBound @Word == 0`.

Are we even using this function? I think integral parsing is done in CLI in a different method. 

I'd strongly suggest to remove this function.